### PR TITLE
Input value restore

### PR DESCRIPTION
**Pull request by mashhoodr**

### CODE REVIEW

File: src/app/shared/form-field/editable.directive.spec.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        fixture.detectChanges();
+        const directiveEl = fixture.debugElement.query(By.css('[editable]'));
+        directiveEl.nativeElement.value = 'testing';
+        directiveEl.nativeElement.focus();
+
+        expect(directiveEl.nativeElement.value).toEqual('');
+      });
+  }));
+
+  it('input value should be restored on blur', inject([TestComponentBuilder], (testComponentBuilder: TestComponentBuilder) => {
```
Comment (mashhoodr) [inline src/app/shared/form-field/editable.directive.spec.ts]: @ZenabKhan can we also have a test checking that the value does not change if the field is not empty

File: src/app/shared/form-field/editable.directive.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+@Directive({
+  selector: '[editable]',
+  host: {
+    '(focus)': '_onFocus()',
+    '(blur)': '_onBlur()',
+  }
+})
+export class EditableDirective {
+  inputValue: any;
```
Comment (mashhoodr) [inline src/app/shared/form-field/editable.directive.ts]: I think input values are always string?

File: src/app/shared/form-field/form-field.component.html [hunk 3]
```diff
@@ -24,7 +24,7 @@
     class="form-control"
     (focus)="onTouched()"
     (blur)="updateValue($event)"
-    placeholder="{{ field[0].placeholder }}" />
+    placeholder="{{ field[0].placeholder }}" editable/>
```
Comment (mashhoodr) [inline src/app/shared/form-field/form-field.component.html]: consider renaming attribute (and class) maybe focus-reset-field? Also put it on the number field?

File: src/app/shared/form-field/editable.directive.spec.ts (hunk 1 above)
Comment (ZenabKhan) [inline src/app/shared/form-field/editable.directive.spec.ts]: The following test does indirectly checks that the value is not changed. Or do I need write a separate test for that?

File: src/app/shared/form-field/form-field.component.html (hunk 3 above)
Comment (ZenabKhan) [inline src/app/shared/form-field/form-field.component.html]: Pushed the renamed attribute and class

File: src/app/shared/form-field/editable.directive.ts (hunk 2 above)
Comment (ZenabKhan) [inline src/app/shared/form-field/editable.directive.ts]: Changed to string.

File: src/app/shared/form-field/editable.directive.spec.ts (hunk 1 above)
Comment (mashhoodr) [inline src/app/shared/form-field/editable.directive.spec.ts]: Better to have an explict test. Testing is also considered to be documentation of your component all functionality should be obvious from it
